usr/src/lib/libzfs/common/libzfs_util.c

```diff
@@ -1250,6 +1252,33 @@ zprop_parse_value(libzfs_handle_t *hdl, nvpair_t *elem, int prop,
 		    prop == ZFS_PROP_SNAPSHOT_LIMIT)) {
 			*ivalp = UINT64_MAX;
 		}
+
+		/*
+		 * Special handling for setting 'reservation' and
+		 * 'refreservation' to 'auto'.  Use UINT64_MAX to tell the
```

Comment:
`reservation=auto` sets it to an amount that's intended to be volsize+metadata, thus preventing a write from failing.  But if there are snapshots, they will consume the reservation and then the write could fail. So I wonder if we really want to support `reservation=auto`.  I don't think there's a valid use case aside from version<9 pools which don't support refreservation, and at this point I think it's reasonable to expect folks to upgrade to version 9 or later if they want the latest features.  I think we risk confusion / accidental misuse if we make it easy to set a reservation that probably doesn't do what they want.

Reply:
I'm happy to not support reservation=auto.  I'm not so sure I want to deal with the bug report from someone that is using zpool version 8 or earlier.  :)